core/src/main/scala-3/medeia/generic/GenericEncoderInstances.scala

```diff
@@ -1,13 +1,51 @@
 package medeia.generic
 
+import shapeless3.deriving.K0.*
 import shapeless3.deriving.*
+import org.mongodb.scala.bson.*
+import scala.compiletime.*
+import medeia.encoder.BsonEncoder
+import scala.deriving.Mirror
 
-trait GenericEncoderInstances {
-  @SuppressWarnings(Array("org.wartremover.warts.ImplicitConversion"))
-  implicit inline def genericEncoder[A](using
-      gen: K0.Generic[A],
-      gdOptions: GenericDerivationOptions[A] = GenericDerivationOptions[A](),
-      stOptions: SealedTraitDerivationOptions[A] = SealedTraitDerivationOptions[A]()
+private[medeia] trait GenericEncoderInstances {
+  inline def apply[A]: GenericEncoder[A] =
+    summonInline[GenericEncoder[A]]
+
+  @SuppressWarnings(Array("org.wartremover.warts.SeqApply"))
+  given coproduct[A](using
+      inst: => K0.CoproductInstances[ProductEncoder, A],
```

Comment:
From here downwards the changes in this file ares mostly code moved from the deleted files. I added `ProductEncoder` and it's usage but the rest mostly stayed the same.